cli/azd/pkg/project/framework_service_npm.go

```diff
@@ -126,11 +126,23 @@ func (np *npmProject) Package(
 				packageSource = filepath.Join(serviceConfig.Path(), serviceConfig.OutputPath)
 			}
 
+			if entries, err := os.ReadDir(packageSource); err != nil || len(entries) == 0 {
+				task.SetError(
+					fmt.Errorf(
+						//nolint:lll
+						"package source '%s' is empty or does not exist. If your service has custom packaging requirements create an NPM script named 'package' within your package.json and ensure your package artifacts are written to the '%s' directory",
```

Comment:
> ensure your package artifacts are written to the '%s' directory

This sounds like `azd` asking customers to update their application in a way azd understands it. I don't think that's a good thing.
azd is usually asking customers where to find things, and not mandating where to and how to do things to work.

Reply:
Can you help me understand why are we doing this?
IIRC, in the past, as a developer, I would just needed to set the path to my application and the path for the `dist` folder where it produces the build artifact.  
Looks like we are changing this to now mandate what folder to use if using azd?  or I might be confused

Reply:
In PR @ https://github.com/Azure/azure-dev/pull/1771 we changed the NPM script we call from build to package to align with our internal alignment of build vs package.

All of our templates have frontend set to dist: build which is the default output directory of the react-scripts build command.

Flow for building our frontend:

1. Run npm run package => Runs react-scripts build and static site is generated into /src/web/build folder
2. We zip up this folder (for app service)
3. Deploy zip to Azure app service
 
Node / NPM does not have any strict requirements on how to build node apps.  There are certain pockets of conventions used by different cohorts of developers but nothing that we can rely on.

In the future I would like to extend the azure.yaml schema to support customizable NPM options (similar to docker options) where the user can tell us what NPM scripts to run for different parts of the azd lifecycle.

Reply:
> In PR @ https://github.com/Azure/azure-dev/pull/1771 we changed the NPM script we call from build to package to align with our internal alignment of build vs package.

I see..  Well,, this is basically telling customers that they must use a script called `package` to run `build` so they are compatible with azd.
Instead of making azd to call `npm buil` for `azd package`, we are extending the concepts to `package` from azd to be the rule for node applications.

This means, for example, that if I run `npx create-react-app someSampleApp` to create a quick empty template for a new web app, it cannot work for azd, as it has not a `package` script. The default scripts are:

![image](https://user-images.githubusercontent.com/24213737/229947430-9471c765-5f5b-46ec-995c-8f3167a17acf.png)

If I try to azdfy this simple app, or copy it inside an existing azd project, I must create a new script like `"package": "react-scripts build"` to make it azd-compatible...  as azd will call `npm run package` to any node app....

That's where I believe we are mandating a new behavior to Node Apps...

Reply:
Took a quick look to the azd-template on github: https://github.com/topics/azd-templates

There are many templates which will be affected by this change (beyond the `todo` samples). For example, a few here:

https://github.com/glaucia86/azure-sql-prisma-vue/blob/main/client/package.json#L5
https://github.com/Azure-Samples/pubsub-dapr-nodejs-servicebus/blob/main/checkout/package.json#L7
https://github.com/Azure-Samples/bindings-dapr-nodejs-cron-postgres/blob/main/batch/package.json#L10
https://github.com/Azure-Samples/app-service-javascript-sap-cloud-sdk-quickstart/blob/main/src/api/package.json#L8

Reply:
Differences: It needs to be in `azure.yaml`, and not in build scripts. For `react` and `vuejs` (using vue-cli), the conventional script is already called `build`. Personally, I would not lean towards the application referring to the publishing step as `package`. It is confusing for multiple reasons, not to say the least that package managers exists. `build` is better in my eyes.

Reply:
Reverted for `build` for the time being based on team discussion.